### DISCUSSION TITLE
want to understand is getLogs [fromBlock, toBlock] both side inclusive? like a SQL clause blockNumber BETWEEN fromBlock AND toBlock ?

### DISCUSSION THREAD

**Discussion by tx0c** [wevm/viem#2263] [Question]
there's an example script, says from `[blockNumber-20n, blockNumber]` is getting events from last 20 blocks, but if both sides are inclusive, then it should be last 21 blocks?
https://github.com/wevm/viem/blob/main/examples/logs_event-logs/index.ts#L15-L21

```js
// Get initial event logs (last 20 blocks)
const blockNumber = await client.getBlockNumber()
const logs = await client.getLogs({
  event,
  fromBlock: blockNumber - 20n,
  toBlock: blockNumber,
})
```

the doc line
https://viem.sh/docs/actions/public/getLogs#toblock is not clear either,

could someone make a doc fix to clarify on this?   Thanks, :tada:

**Comment by JoelAlexander** [comment]
I think you can be confident it's inclusive to both.  Would be nice if the docs were explicit

https://ethereum.stackexchange.com/questions/78620/is-the-range-of-a-filterquery-inclusive-up-to-toblock
